snuba/datasets/eventsv2.py

```diff
@@ -0,0 +1,75 @@
+from datetime import timedelta
+from typing import Sequence
+
+from snuba.datasets import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.factory import get_dataset
+from snuba.datasets.schemas.join import (
+    JoinConditionExpression,
+    JoinCondition,
+    JoinedSchema,
+    JoinStructure,
+    JoinType,
+    SchemaJoinedSource,
+    SubJoinSource,
+)
+from snuba.query.extensions import PERFORMANCE_EXTENSION_SCHEMA, PROJECT_EXTENSION_SCHEMA
+from snuba.query.schema import GENERIC_QUERY_SCHEMA
+from snuba.request import RequestSchema
+from snuba.schemas import get_time_series_extension_properties
+
+
+class EventsV2(TimeSeriesDataset):
+    def __init__(self) -> None:
+        grouped_message = get_dataset("groupedmessage")
+        events = get_dataset("events")
+
+        join_structure = JoinStructure(
+            left_source=SchemaJoinedSource(
+                "groups",
+                grouped_message.get_dataset_schemas().get_read_schema(),
+            ),
+            right_source=SchemaJoinedSource(
+                "events",
+                events.get_dataset_schemas().get_read_schema(),
+            ),
+            mapping=[
+                JoinCondition(
+                    left=JoinConditionExpression(table_alias="groups", column="project_id"),
+                    right=JoinConditionExpression(table_alias="events", column="project_id"),
+                ),
+                JoinCondition(
+                    left=JoinConditionExpression(table_alias="groups", column="id"),
+                    right=JoinConditionExpression(table_alias="events", column="group_id"),
+                ),
+            ],
+            join_type=JoinType.LEFT,
```

Comment:
Will revisit the type of join when we will start building the event search on top of this.